src/coroutine.js

```diff
@@ -25,6 +26,36 @@ export default function coroutine (action, body, repo) {
     return action
   }
 
+  if (body && get(body, 'constructor.name') === 'GeneratorFunction') {
+    action.open()
+
+    let iterator = body(repo.push.bind(repo), repo)
```

Comment:
So you can reference the repo. I think just passing repo in here makes a ton of sense.